### DISCUSSION TITLE
Vector Shape Properties for Lighting

### DISCUSSION THREAD

**Discussion by diegostoliar** [keiwando/toonsquid#476] [Ideas]
Some softwares and some new pipelines offer a set of tools to shape and tween lighting shapes. 

Toonboom offers Animated Matte Generators:
https://docs.toonboom.com/help/harmony-15/premium/effects/animated-matte-generator/about-animated-matte-generator.html
which can be taken really far, we are being able to create some cool results with this tech at the studio here.

Other studios create their own solutions, like Sergio Pablos Studio did for Klauss:
https://www.youtube.com/watch?v=nTcblxN8TGQ

I was wondering: what would be the minimum that needs to be added to Toon Squid to achieve something similar?
Toon Squid already has vector shapes that can be animated, masked, blended and colored. It also has line color and width. I guess 2 things that could be missing are:

- Line Pressure on vectors
- Feathering (Blur) on vectors

![ShapeProperties1](https://github.com/keiwando/toonsquid/assets/10113975/f8ec2b82-d1be-4b8e-8be5-eebff1806801)
if we had Feathering (sorry about the typo on the image) and Point Pressure on the shape properties, we would be able to make shapes like these:
![ShapeProperties2](https://github.com/keiwando/toonsquid/assets/10113975/b0ce5035-45a0-465b-a2e0-34c500e453b6)
Which would be also cool for other things, like making a vector character with line thickness etc... But for lighting this could be use to choke the shape and make shadow creases, which are very important to represent 3d volume, and we can use the color of the line to do light terminators.
![ShapeProperties3](https://github.com/keiwando/toonsquid/assets/10113975/9f69efa7-8557-492d-83b2-d93d49e9e9d3)
Than we could increase the feathering (blur) or each shape to shade a character. This could also be achieved with non-detructive blur added to the shape layer, like the one we see on Procreate Dreams.

There is definetily more to the lighting process, but this would be a good start, you would be able to do a lot with that.

**Comment by keiwando** [comment]
Both variable-width strokes and non-destructive blur are features that I want to add in the future.

**Reply by diegostoliar** [reply]
Animatable line weight? That would be supercool!

**Reply by DWrightAnimation** [reply]
I'll say. Imagine doing puppet rigged animation, but you can create the illusion of imperfection in the lines to make it look hand drawn. This line boiling is something Ed, Edd, N Eddy used in its style, that's why it looked like the characters were never on a hold, because a single pose would have 2 or more variations for the line art.

**Reply by DWrightAnimation** [reply]
It's also be cool if we could have textured lines, instead of being limited to simple clean lines. Think about the show Hilda, how the line art looks grainy like chalk or crayon, imagine being able to animate curves with those kind of lines.
